doc/lua_code_style_guidelines.md

```diff
@@ -0,0 +1,330 @@
+
+Note that these are only *guidelines* for more readable code.  In some (rare) cases they may result in *less* readable code.  Use your best judgement.
+
+
+Comments
+--------
+
+* Incorrect or outdated comments are worse than no comments.
+
+* Avoid inline comments, unless they're very short.
+
+* Write comments to clarify anything that may be confusing. Don't write comments that describe things that are obvious from the code.
+
+<span style="color: #282;">Good:</span>
+```lua
+width = width - 2  -- Adjust for 1px border on each side
+```
+
+<span style="color: #b44;">Bad:
+```lua
+width = width - 2  -- Decrement width by two
+```
+
+* Comments should follow English grammar rules, this means **starting with a capital letter**, using commas and apostrophes where appropriate, and **ending with a period**. The period may be omitted in single-sentence comments. If the first word of a comment is an identifier, its casing should not be changed. **Check the spelling.**
+```lua
+-- This is a properly formatted comment.
+-- This isnt.              (missing apostrophe)
+-- neither is this.        (lowercase first letter)
+```
+
+* Comments should have a space between the comment characters and the first word.
+```lua
+--This is wrong.
+-- This is right.
+```
+
+* Inline comments should be separated from the code by two spaces.
+```lua
+foo()  -- A proper comment.
+```
+
+* If you write comments for a documentation generation tool, write the comments in LuaDoc format.
+
+* Short multi-line comments should use the regular single-line comment style.
+
+* Long multi-line comments should use Lua's multi-line comment format with no leading characters except a `--` before the closer.
+```lua
+--[[
+Very long
+multi-line comment.
+--]]
+```
+
+
+Lines, spaces, and indentation
+------------------------------
+
+* Indentation is done with one tab per indentation level.
+
+* **Lines are wrapped at 80 characters** where possible, with a hard limit of 90.  If you need more you're probably doing something wrong.
+
+
+# Continuation lines
+
+* Conditional expressions have continuation lines indented by two tabs.
+```lua
+if long_function_call(with, many, arguments) and
+		another_function_call() then
+	do_something()
+end
+```
+
+* Function arguments are indented by two tabs if multiple arguments are in a line, same for definition continuations.
+```lua
+foo(bar, biz, "This is a long string..."
+		baz, qux, "Lua")
+
+function foo(a, b, c, d,
+		e, f, g, h)
+	[…]
+end
+```
+
+* If the function arguments contain a table, it's indented by one tab and if the arguments get own lines, it's indented like a table.
+```lua
+register_node("name", {
+	"This is a long string...",
+	0.3,
+})
+
+list = filterlist.create(
+	preparemodlist,
+	comparemod,
+	function()
+		return "no comma at the end"
```

Comment:
A comma at the end wouldn't be allowed syntactically.

Reply:
Correct.
This example doesn't hurt anyway though, imo.